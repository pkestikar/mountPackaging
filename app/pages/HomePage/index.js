import React from 'react';
import Button from "Components/Button";
import Icon from 'Components/Icon';
import { makeStyles } from '@material-ui/core/styles';
import Grid from '@material-ui/core/Grid';
import { IMAGE_PATH } from "../../constants";
import Container from '@material-ui/core/Container';

const ServiceInfoItemComposer = (props) => {
    return (
        <Grid container spacing={2}>
            <Grid item xs={12} md={4}>
                <img className={'service-info-img'} src={`${IMAGE_PATH}${props.image}.png`} alt={props.image}/>
            </Grid>
            <Grid item xs={12} md={8}>
                <div className="service-info-description">
                    <h3>{props.title}</h3>
                    <div className="text">
                        {props.description}
                    </div>
                </div>
            </Grid>
        </Grid>
    )
};

const HomePage = (props) => {
    const serviceItems = [
        {
            image: 'paint-bucket',
            title: '01 PAINT',
            description: 'We have experience in manufacturing packaging solutions for a gamut of containers with various sizes, for any of the volumetric or gravimetric requirements.'
        },
        {
            image: 'oil-bucket',
            title: '02 OILS',
            description: 'We provide packaging solutions for all kinds of oils like gear oils, lubricant oils, edible oils, etc.'
        },
        {
            image: 'ink-bottle',
            title: '03 INKS',
            description: 'While striving for an efficient solution, our packaging machinery designs for inks ensure least amount of contamination.'
        },
        {
            image: 'barrel',
            title: '04 CHEMICALS',
            description: 'We are able to provide packaging machinery for chemicals like peroxides with Lance bottom and top filling.'
        },
        {
            image: 'adhesive-tube',
            title: '05 ADHESIVES',
            description: 'We can provide packaging solutions for adhesives  with volumetric/pressure type filling.'
        }
    ];

    return (
        <div className="mp--homepage-container">
            <div className="mp--homepage--first-section">
                <div className={'background-curve'}></div>
                <div className="mp--homepage--first-section-info-container">
                    <div className="introduction-text">
                        <span className="we-design">WE DESIGN.</span>
                        <span className="we-deliver">WE DELIVER.</span>
                    </div>
                    <div className="introduction-description">
                        <span className={'description-text'}>Tailored to the customer requirements, we design and deliver top-notch packaging machinery.</span>
                    </div>
                    <div className="introduction-cta">
                        <Button palette={'primary'}>Let's Connect</Button>
                    </div>
                    <div className="scroll-down-indicator">
                        <Icon icon={'scroll'} />
                    </div>
                </div>
            </div>
            <div className="mp--homepage--second-section">
                <Container maxWidth={'lg'}>
                    <div className="second-section-heading">
                        <h2>WE SPECIALIZE IN</h2>
                    </div>
                    {
                        serviceItems.map(item => {
                            return (
                                <ServiceInfoItemComposer
                                    image={item.shape}
                                    title={item.title}
                                    description={item.description}
                                />
                            )
                        })
                    }
                </Container>
            </div>
        </div>
    )
};

export default HomePage;
