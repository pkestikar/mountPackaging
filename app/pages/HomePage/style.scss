.mp--homepage-container {
  .mp--homepage--first-section {
    position: relative;
    height: 100vh;
    display: flex;
    justify-content: center;
    align-items: center;
    .background-curve {
      position: absolute;
      top: 0;
      right: 0;
      max-width: 50%;
      width: 100%;
      height: 100%;
      background: linear-gradient(to bottom, $navy-blue, $light-periwinkle);
      clip-path: ellipse(85.7% 60% at 86% 50%);
      z-index: -5;
    }

    &-info-container {
      display: flex;
      justify-content: center;
      align-items: center;
      flex-direction: column;
        .introduction-text {
          display: flex;
          width: 600px;
          max-width: 100vw;
          span {
            font-size: 2.8rem;
            @include breakpoint(small down) {
              font-size: 2rem;
            }
            font-weight: bold;
            text-align: right;
            width: 50%;
          }
          .we-design {
            color: var(--navy-blue);
          }
          .we-deliver {
            color: var(--light-periwinkle);
            margin-left: 1rem;
            @include breakpoint(small down) {
              text-align: left;
            }
          }
        }
      .introduction-description {
        margin: 1rem;
        text-align: center;
      }
      .introduction-cta {
        margin: 1rem;
      }
      .scroll-down-indicator {
        position: absolute;
        bottom: 10px;
        img, svg {
          height: 36px !important;
          width: 36px !important;
        }
      }
      //}
    }
  }

  .mp--homepage--second-section {
    background: var(--navy-blue);
    width: 100%;
    .second-section-heading {
      text-align: center;
      padding: 3rem 0;
      color: white;
      font-size: 1.5rem;
      @media only screen and (max-width: $medium) {
        font-size: 1.2rem;
        padding: 3rem 0 1rem;
      }
    }

    .service-info-item-container {
      .service-info-img-wrapper {
        display: flex;
        justify-content: center;
        align-items: center;
      }
      .service-info-description {
        height: 260px;
        color: white;
        .title {
          font-size: 1.8rem;
          margin-top: 3rem;
        }
        .text {
          margin-top: 1.3rem;
          line-height: 1.8;
        }
      }

      &__mobile {
        height: 397px;
        display: flex;
        flex-direction: column;
        align-items: center;
        color: white;
        max-width: 285px;
        border: 1px solid white;
        margin: 2rem auto;
        padding: 0 1rem;
        .service-info-img-wrapper {
          margin-top: 3rem;
          img {
            max-height: 110px;
          }
        }
        .service-info-description {
          .title {
            font-size: 1.5rem;
            margin: 1rem 0;
            text-align: center;
          }
          .text {
            font-size: 0.9rem;
          }
        }
      }
    }

    .swiper-container {
      .swiper-pagination-bullets {
        bottom: 50px;
      }
    }
  }

  .mp--homepage--third-section {
    height: 350px;
    display: flex;
    justify-content: center;
    align-items: center;
    flex-direction: column;
    span {
      font-size: 1.8rem;
      margin-bottom: 2rem;
      text-align: center;
    }
    @media only screen and (max-width: $medium) {
      height: 200px;
      span {
        font-size: 1.4rem;
        margin-bottom: 0.9rem;
      }
    }
  }
}
