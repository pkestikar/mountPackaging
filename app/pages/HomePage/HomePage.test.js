import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import HomePage from './index';
import { serviceItems, ServiceInfoItemComposer } from "./index";

describe('HomePage component', () => {
    let wrapper;
    beforeEach(() => {
        wrapper = shallow(<HomePage />);
    });

    it('should render correctly', () => {
        expect(toJSON(wrapper)).toMatchSnapshot();
    });

    it('should have 5 service elements in the WE SPECIALIZE IN section', () => {
        expect(wrapper.find('.mp--homepage--second-section').childAt(1).find('ServiceInfoItemComposer').length).toEqual(5);
    });

    it('should have the right info passed in for the 5 service info elements', () => {
        wrapper.find('.mp--homepage--second-section').childAt(1).find('ServiceInfoItemComposer').forEach((node, index) => {
            expect(node.prop('image')).toEqual(serviceItems[index].image);
            expect(node.prop('title')).toEqual(serviceItems[index].title);
            expect(node.prop('description')).toEqual(serviceItems[index].description);
        });
    });
});

describe('ServiceInfoItemComposer component', () => {
    let wrapper;
    beforeEach(() => {
        wrapper = shallow(<ServiceInfoItemComposer
                image={serviceItems[0].image}
                title={serviceItems[0].title}
                description={serviceItems[0].description}
            />
        );
    });

    it('should render correctly', () => {
        expect(toJSON(wrapper)).toMatchSnapshot();
    });

    it('should set the image path correctly', () => {
        expect(wrapper.find('.service-info-img').prop('src')).toContain(serviceItems[0].image);
    });


    it('should set the title correctly', () => {
        expect(wrapper.find('.title').text()).toEqual(serviceItems[0].title);
    });

    it('should set the description correctly', () => {
        expect(wrapper.find('.text').text()).toEqual(serviceItems[0].description);
    });
});
