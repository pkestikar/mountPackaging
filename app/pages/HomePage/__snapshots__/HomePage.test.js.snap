// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage component should render correctly 1`] = `
<div
  className="mp--homepage-container"
>
  <div
    className="mp--homepage--first-section"
  >
    <div
      className="background-curve"
    />
    <div
      className="mp--homepage--first-section-info-container"
    >
      <div
        className="introduction-text"
      >
        <span
          className="we-design"
        >
          WE DESIGN.
        </span>
        <span
          className="we-deliver"
        >
          WE DELIVER.
        </span>
      </div>
      <div
        className="introduction-description"
      >
        <span
          className="description-text"
        >
          Tailored to the customer requirements, we design and deliver top-notch packaging machinery.
        </span>
      </div>
      <div
        className="introduction-cta"
      >
        <Button
          disabled={false}
          loading={false}
          palette="primary"
          size="medium"
        >
          Let's Connect
        </Button>
      </div>
      <div
        className="scroll-down-indicator"
      >
        <Icon
          icon="scroll"
          onClick={[Function]}
          sizeInPixels="16"
        />
      </div>
    </div>
  </div>
  <div
    className="mp--homepage--second-section"
  >
    <div
      className="second-section-heading"
    >
      WE SPECIALIZE IN
    </div>
    <MQMinLarge>
      <WithStyles(ForwardRef(Container))
        disableGutters={true}
        maxWidth="lg"
      >
        <ServiceInfoItemComposer
          description="We have experience in manufacturing packaging solutions for a gamut of containers with various sizes, for any of the volumetric or gravimetric requirements."
          image="paint-bucket"
          key="service-info-item-0"
          title="01 PAINTS"
        />
        <ServiceInfoItemComposer
          description="We provide packaging solutions for all kinds of oils like gear oils, lubricant oils, edible oils, etc."
          image="oil-bucket"
          key="service-info-item-1"
          title="02 OILS"
        />
        <ServiceInfoItemComposer
          description="While striving for an efficient solution, our packaging machinery designs for inks ensure least amount of contamination."
          image="ink-bottle"
          key="service-info-item-2"
          title="03 INKS"
        />
        <ServiceInfoItemComposer
          description="We are able to provide packaging machinery for chemicals like peroxides with Lance bottom and top filling."
          image="barrel"
          key="service-info-item-3"
          title="04 CHEMICALS"
        />
        <ServiceInfoItemComposer
          description="We can provide packaging solutions for adhesives  with volumetric/pressure type filling."
          image="adhesive-tube"
          key="service-info-item-4"
          title="05 ADHESIVES"
        />
      </WithStyles(ForwardRef(Container))>
    </MQMinLarge>
    <MQMaxMedium>
      <Swiper
        navigation={false}
        pagination={
          Object {
            "clickable": true,
          }
        }
        slidesPerView={1}
      >
        <SwiperSlide
          key="service-0"
        >
          <div
            className="service-info-item-container__mobile"
          >
            <div
              className="service-info-img-wrapper"
            >
              <img
                alt="paint-bucket"
                className="service-info-img"
                src="/images/paint-bucket.png"
              />
            </div>
            <div
              className="service-info-description"
            >
              <div
                className="title"
              >
                01 PAINTS
              </div>
              <div
                className="text"
              >
                We have experience in manufacturing packaging solutions for a gamut of containers with various sizes, for any of the volumetric or gravimetric requirements.
              </div>
            </div>
          </div>
        </SwiperSlide>
        <SwiperSlide
          key="service-1"
        >
          <div
            className="service-info-item-container__mobile"
          >
            <div
              className="service-info-img-wrapper"
            >
              <img
                alt="oil-bucket"
                className="service-info-img"
                src="/images/oil-bucket.png"
              />
            </div>
            <div
              className="service-info-description"
            >
              <div
                className="title"
              >
                02 OILS
              </div>
              <div
                className="text"
              >
                We provide packaging solutions for all kinds of oils like gear oils, lubricant oils, edible oils, etc.
              </div>
            </div>
          </div>
        </SwiperSlide>
        <SwiperSlide
          key="service-2"
        >
          <div
            className="service-info-item-container__mobile"
          >
            <div
              className="service-info-img-wrapper"
            >
              <img
                alt="ink-bottle"
                className="service-info-img"
                src="/images/ink-bottle.png"
              />
            </div>
            <div
              className="service-info-description"
            >
              <div
                className="title"
              >
                03 INKS
              </div>
              <div
                className="text"
              >
                While striving for an efficient solution, our packaging machinery designs for inks ensure least amount of contamination.
              </div>
            </div>
          </div>
        </SwiperSlide>
        <SwiperSlide
          key="service-3"
        >
          <div
            className="service-info-item-container__mobile"
          >
            <div
              className="service-info-img-wrapper"
            >
              <img
                alt="barrel"
                className="service-info-img"
                src="/images/barrel.png"
              />
            </div>
            <div
              className="service-info-description"
            >
              <div
                className="title"
              >
                04 CHEMICALS
              </div>
              <div
                className="text"
              >
                We are able to provide packaging machinery for chemicals like peroxides with Lance bottom and top filling.
              </div>
            </div>
          </div>
        </SwiperSlide>
        <SwiperSlide
          key="service-4"
        >
          <div
            className="service-info-item-container__mobile"
          >
            <div
              className="service-info-img-wrapper"
            >
              <img
                alt="adhesive-tube"
                className="service-info-img"
                src="/images/adhesive-tube.png"
              />
            </div>
            <div
              className="service-info-description"
            >
              <div
                className="title"
              >
                05 ADHESIVES
              </div>
              <div
                className="text"
              >
                We can provide packaging solutions for adhesives  with volumetric/pressure type filling.
              </div>
            </div>
          </div>
        </SwiperSlide>
      </Swiper>
    </MQMaxMedium>
  </div>
  <div
    className="mp--homepage--third-section"
  >
    <span>
      Ready with requirements?
    </span>
    <Button
      disabled={false}
      loading={false}
      palette="secondary"
      size="medium"
    >
      Get Quotation
    </Button>
  </div>
</div>
`;

exports[`ServiceInfoItemComposer component should render correctly 1`] = `
<WithStyles(ForwardRef(Grid))
  className="service-info-item-container"
  container={true}
  spacing={2}
>
  <WithStyles(ForwardRef(Grid))
    className="service-info-img-wrapper"
    item={true}
    md={4}
    xs={12}
  >
    <img
      alt="paint-bucket"
      className="service-info-img"
      src="/images/paint-bucket.png"
    />
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    className="service-info-description"
    item={true}
    md={8}
    xs={12}
  >
    <div
      className="title"
    >
      01 PAINTS
    </div>
    <div
      className="text"
    >
      We have experience in manufacturing packaging solutions for a gamut of containers with various sizes, for any of the volumetric or gravimetric requirements.
    </div>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;
